server/network.go

```diff
@@ -149,7 +154,14 @@ func newLocalNetwork(opts localNetworkOptions) (*localNetwork, error) {
 // TODO document.
 // Assumes [lc.lock] is held.
 func (lc *localNetwork) createConfig() error {
-	cfg, err := local.NewDefaultConfigNNodes(lc.options.execPath, lc.options.numNodes, lc.options.networkID)
+	cfg, err := local.NewDefaultConfigNNodes(
+		lc.options.execPath,
+		lc.options.numNodes,
+		lc.options.networkID,
+		lc.options.genesisPath,
```

Comment:
for a default network, it may be receiving too specific stuff. probably. only genesis is needed, cfg.Upgrade to be set afterwars, outside of this call, and instead of beacon config, a boolean flag indicating that the validators live outside of this network.

Reply:
this one I don't agree with. Upgrade config is essential and should be treated as genesis, IMHO
no point of additional boolean flag - if beacon.Set is not null, it should be used right away

Reply:
anyway, I can move it from default into this func
and see how it will look like

Reply:
pls disregard my previus comment.